src/sagas/repository.ts

```diff
@@ -545,6 +545,29 @@ export function* getDatasetTablePreview({ payload }: any): any {
   }
 }
 
+/**
+ * Jobs.
+ */
+
+export function* getJobs({ payload }: any): any {
+  const { limit, offset, sort, direction, searchString } = payload;
+  try {
+    const response = yield call(
+      authGet,
+      `/api/repository/v1/jobs?offset=${offset}&limit=${limit}&sort=${sort}&direction=${direction}&filter=${searchString}`,
+    );
+    yield put({
+      type: ActionTypes.GET_JOBS_SUCCESS,
+      jobs: { data: response },
+    });
+  } catch (err) {
+    showNotification(err);
```

Comment:
Nice work on adding a generator!  This is nice and clean.

One small tweak to consider: instead of popping a toast with technical details, I've started to include messages to inform the user what to do.  For example, showNotification could pop: 
`An error occurred loading jobs.  Reload the page to try again.`
In cases where these generators are used by multiple components, it might be useful to pass in the error message with the action parameters.
If we want the browser to capture technical details of the error, we could log those to the console.

Reply:
Mm, I like that, the users often dont need to see the detailed error message. I think this sort of error handling should be done by the component that is calling the method in that case, since the solution could change based on how this method is being called.

For example, instead of refreshing the page, maybe the method is being called via button click. This repository should not know how they are being called or what the error handling would need to be - that is the responsibility of the caller.

What do you think of that?

Reply:
Exactly where I was going.  This juice is worth the squeeze.  :-)

Reply:
Updated this so you can pass in an error message - I kind of .... went crazy at first, and was modifying the repository to allow for an onError method to be passed in initially. I think it might be the better long term solution, but I feel a little unconfident about it, like, maybe it would be something better to add if we eventually find the need for it? But I was designing something far too complex, like "what if we wanted x to happen on this type of error, but y to happen on this other type of error"

I think ultimately, if getJobs failed, it's because *we did something wrong, not the user, and in this one specific instance, they probably dont need something too robust!